### ISSUE TITLE
Daemon stopped kwarg not being set causing never ending hung process

### ISSUE THREAD

**Issue by jeff-minard-ck** [nolar/kopf#852]
### Long story short

I did a little binary search and found that this code works in 1.34.0 and then creates a never ending loop in 1.35.0 (also present in 1.35.1). The `stopped` kwarg seems to no longer be getting set at stop time.

### Kopf version

1.35.0

### Kubernetes version

kind 1.22

### Python version

3.9

### Code

```python
import kopf
import time

@kopf.on.daemon('apps', 'v1', 'deployments')
def my_func(stopped, namespace, name, **kwargs):
    while not stopped:
        print(f"loop {namespace} / {name}")
        print(stopped)
        print(stopped.is_set())
        time.sleep(1.0)
    print("We are done. Bye.")
    return
```

### Logs

```none
$ kopf run --standalone --log-format=json tod.py -v -A
{"message": "Starting Kopf 1.35.0.", "timestamp": "2021-10-26T17:06:47.440629+00:00", "severity": "debug"}
{"message": "Initial authentication has been initiated.", "timestamp": "2021-10-26T17:06:47.440954+00:00", "severity": "info"}
{"message": "Activity 'login_via_pykube' is invoked.", "timestamp": "2021-10-26T17:06:47.441114+00:00", "severity": "debug"}
{"message": "Pykube is configured in cluster with service account.", "timestamp": "2021-10-26T17:06:47.464331+00:00", "severity": "debug"}
{"message": "Activity 'login_via_pykube' succeeded.", "timestamp": "2021-10-26T17:06:47.469164+00:00", "severity": "info"}
{"message": "Activity 'login_via_client' is invoked.", "timestamp": "2021-10-26T17:06:47.469380+00:00", "severity": "debug"}
{"message": "Client is configured in cluster with service account.", "timestamp": "2021-10-26T17:06:47.485628+00:00", "severity": "debug"}
{"message": "Activity 'login_via_client' succeeded.", "timestamp": "2021-10-26T17:06:47.486409+00:00", "severity": "info"}
{"message": "Initial authentication has finished.", "timestamp": "2021-10-26T17:06:47.488600+00:00", "severity": "info"}
{"message": "Starting the watch-stream for customresourcedefinitions.v1.apiextensions.k8s.io cluster-wide.", "timestamp": "2021-10-26T17:06:47.545344+00:00", "severity": "debug"}
{"message": "Starting the watch-stream for deployments.v1.apps cluster-wide.", "timestamp": "2021-10-26T17:06:47.546418+00:00", "severity": "debug"}
{"message": "Stopping the watch-stream for customresourcedefinitions.v1.apiextensions.k8s.io cluster-wide.", "timestamp": "2021-10-26T17:06:47.551543+00:00", "severity": "debug"}
{"message": "Not enough permissions to watch for resources: changes (creation/deletion/updates) will not be noticed; the resources are only refreshed on operator restarts.", "timestamp": "2021-10-26T17:06:47.552145+00:00", "severity": "warn"}
{"message": "Daemon 'my_func' is invoked.", "timestamp": "2021-10-26T17:06:47.668264+00:00", "object": {"apiVersion": "apps/v1", "kind": "Deployment", "name": "example-tod", "uid": "8eb8e192-5c21-43f8-a782-cf683ae6c3df", "namespace": "default"}, "severity": "debug"}
{"message": "Daemon 'my_func' is invoked.", "timestamp": "2021-10-26T17:06:47.668811+00:00", "object": {"apiVersion": "apps/v1", "kind": "Deployment", "name": "coredns", "uid": "a16b3387-2f69-462c-a0b3-39b01e54fe20", "namespace": "kube-system"}, "severity": "debug"}
loop default / example-tod
<FlagSetter: False, reason=None>
False
{"message": "Daemon 'my_func' is invoked.", "timestamp": "2021-10-26T17:06:47.670113+00:00", "object": {"apiVersion": "apps/v1", "kind": "Deployment", "name": "local-path-provisioner", "uid": "95c9cbde-76f5-4eb9-b056-ac746f76969b", "namespace": "local-path-storage"}, "severity": "debug"}
loop kube-system / coredns
{"message": "Daemon 'my_func' is invoked.", "timestamp": "2021-10-26T17:06:47.671470+00:00", "object": {"apiVersion": "apps/v1", "kind": "Deployment", "name": "paas-time-of-day-scaler", "uid": "9fff9107-f08a-45c1-8099-753231fa914f", "namespace": "paas-time-of-day-scaler"}, "severity": "debug"}
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
<FlagSetter: False, reason=None>
False
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
^C{"message": "Signal SIGINT is received. Operator is stopping.", "timestamp": "2021-10-26T17:06:50.034944+00:00", "severity": "info"}
{"message": "Poster of events is cancelled.", "timestamp": "2021-10-26T17:06:50.035288+00:00", "severity": "debug"}
{"message": "Admission validating configuration manager is cancelled.", "timestamp": "2021-10-26T17:06:50.035408+00:00", "severity": "debug"}
{"message": "Resource observer is cancelled.", "timestamp": "2021-10-26T17:06:50.036048+00:00", "severity": "debug"}
{"message": "Admission insights chain is cancelled.", "timestamp": "2021-10-26T17:06:50.036243+00:00", "severity": "debug"}
{"message": "Admission mutating configuration manager is cancelled.", "timestamp": "2021-10-26T17:06:50.036386+00:00", "severity": "debug"}
{"message": "Credentials retriever is cancelled.", "timestamp": "2021-10-26T17:06:50.036512+00:00", "severity": "debug"}
{"message": "Namespace observer is cancelled.", "timestamp": "2021-10-26T17:06:50.036637+00:00", "severity": "debug"}
{"message": "Admission webhook server is cancelled.", "timestamp": "2021-10-26T17:06:50.036874+00:00", "severity": "debug"}
{"message": "Stopping the watch-stream for deployments.v1.apps cluster-wide.", "timestamp": "2021-10-26T17:06:50.037386+00:00", "severity": "debug"}
{"message": "Daemon killer is cancelled.", "timestamp": "2021-10-26T17:06:50.037929+00:00", "severity": "debug"}
{"message": "Streaming tasks are stopped: finishing normally; tasks left: set()", "timestamp": "2021-10-26T17:06:50.139824+00:00", "severity": "debug"}
{"message": "Multidimensional multitasker is cancelled.", "timestamp": "2021-10-26T17:06:50.140151+00:00", "severity": "debug"}
{"message": "Root tasks are stopped: finishing normally; tasks left: set()", "timestamp": "2021-10-26T17:06:50.141161+00:00", "severity": "debug"}
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
{"message": "Hung tasks are not stopped: finishing normally; tasks left: {<Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2aa40c10>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5310>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a54c0>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5550>()]>>}", "timestamp": "2021-10-26T17:06:56.127026+00:00", "severity": "debug"}
loop default / example-tod
<FlagSetter: False, reason=None>
loop local-path-storage / local-path-provisioner
False
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
{"message": "Hung tasks are not stopped: finishing normally; tasks left: {<Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2aa40c10>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5310>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a54c0>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5550>()]>>}", "timestamp": "2021-10-26T17:06:57.129283+00:00", "severity": "debug"}
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
{"message": "Hung tasks are not stopped: finishing normally; tasks left: {<Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2aa40c10>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5310>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a54c0>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5550>()]>>}", "timestamp": "2021-10-26T17:06:58.131921+00:00", "severity": "debug"}
loop local-path-storage / local-path-provisioner
<FlagSetter: False, reason=None>
False
loop default / example-tod
<FlagSetter: False, reason=None>
False
loop paas-time-of-day-scaler / paas-time-of-day-scaler
<FlagSetter: False, reason=None>
False
loop kube-system / coredns
<FlagSetter: False, reason=None>
False
{"message": "Hung tasks are not stopped: finishing normally; tasks left: {<Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2aa40c10>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5310>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a54c0>()]>>, <Task pending name='runner of my_func' coro=<_runner() running at /home/pyoperator/.local/lib/python3.9/site-packages/kopf/_core/engines/daemons.py:409> wait_for=<Future pending cb=[shield.<locals>._outer_done_callback() at /usr/local/lib/python3.9/asyncio/tasks.py:905, <TaskWakeupMethWrapper object at 0x7f9a2a1a5550>()]>>}", "timestamp": "2021-10-26T17:06:59.134182+00:00", "severity": "debug"}
```

**Comment by nolar** [comment]
Thanks for reporting. I can reproduce that. A little note: the daemon is stopped normally when the resource is deleted. The issue is only with the operator termination (Ctrl-C/SIGTERM/etc). I'll take a look now.

**Comment by nolar** [comment]
Also, a side-issue (related or not): it passes the `FlagSetter` type, while it should be the read-only `FlagWaiter`.

**Comment by nolar** [comment]
The fix is released as [1.35.2](https://github.com/nolar/kopf/releases/tag/1.35.2)

**Comment by jeff-minard-ck** [comment]
Confirmed, thanks!
